jupyter_collaboration/loaders.py

```diff
@@ -0,0 +1,94 @@
+import asyncio
+from logging import Logger, getLogger
+from typing import Any, Callable, Coroutine, Dict, Optional
+
+from jupyter_server.utils import ensure_async
+
+
+class FileLoader:
+    def __init__(
+        self,
+        path: str,
+        file_format: str,
+        file_type: str,
+        contents_manager: Any,
+        log: Optional[Logger],
+        poll_interval: Optional[float] = None,
+    ) -> None:
+        self._path: str = path
+        self._file_format: str = file_format
+        self._file_type: str = file_type
+        self._last_modified = None
+
+        self._lock = asyncio.Lock()
+        self._poll_interval = poll_interval
+        self._contents_manager = contents_manager
+
+        self._log = log or getLogger(__name__)
+        self._subscriptions: Dict[str, Callable[[str], Coroutine[Any, Any, None]]] = {}
+
+        if self._poll_interval:
+            self._watcher = asyncio.create_task(self.watch_file())
+
+    @property
+    def path(self):
+        return self._path
+
+    def clean(self) -> None:
+        self._watcher.cancel()
+
+    def rename_file(self, path: str) -> None:
+        self._path = path
```

Comment:
Where is this used?

Reply:
Is not used

Reply:
How do you handle file renames then? Shouldn't it be connected to the file ID service, which knows about files that are renamed?

Reply:
Good catch! I was not handling the renames.
I just pushed a commit to use file ids instead of paths and do the transformation when saving. This fixes the rename.